### DISCUSSION TITLE
Pass `Signal<T>` to a Component from its parent layout

### DISCUSSION THREAD

**Discussion by wackazong** [DioxusLabs/dioxus#2330] [Q&A] [answered]
I am using dioxus_router for routing. I have a situation where I want to pass a `Signal<T>` from the layout down to the child components, but at the same time I want to be able to route to individual child components. Is that possible somehow?

**Comment by ealmloff** [comment, marked as answer]
You can provide the signal through the context API in your layout and consume it in the routes: https://dioxuslabs.com/learn/0.5/reference/context

**Comment by wackazong** [comment]
Yes, I thought so. But I hoped for a more succinct way of doing it. I try to avoid the context API because it makes the code harder to read IMHO.

Thanks for the quick answer!
